pkg/daemon/controller/controller_test.go

```diff
@@ -222,6 +222,118 @@ func TestReconcileService(t *testing.T) {
+		expectedServicesNameCmdPairs map[string]string
+		expectErr                    bool
+	}{
+		{
+			name:                         "No services",
+			configMapServices:            []servicescm.Service{},
+			expectedServicesNameCmdPairs: map[string]string{},
+			expectErr:                    false,
+		},
+		{
+			name: "Single bootstrap service",
+			configMapServices: []servicescm.Service{
+				{
+					Name:         "test1",
+					Command:      "test1 --var=TEST",
+					Dependencies: nil,
+					Bootstrap:    true,
+					Priority:     0,
+				},
+			},
+			expectedServicesNameCmdPairs: map[string]string{"test1": "test1 --var=TEST"},
+			expectErr:                    false,
+		},
+		{
+			name: "No bootstrap services",
+			configMapServices: []servicescm.Service{
+				{
+					Name:         "test1",
+					Command:      "test1 arg1",
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     0,
+				},
+				{
+					Name:         "test2",
+					Command:      "test2 arg1 arg2",
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     1,
+				},
+			},
+			expectedServicesNameCmdPairs: map[string]string{},
+			expectErr:                    false,
+		},
+		{
+			name: "Multiple mixed services",
+			configMapServices: []servicescm.Service{
+				{
+					Name:         "test1",
+					Command:      "test1 arg1",
+					Dependencies: nil,
+					Bootstrap:    true,
+					Priority:     0,
+				},
+				{
+					Name:         "test2",
+					Command:      "test2",
+					Dependencies: nil,
+					Bootstrap:    true,
+					Priority:     1,
+				},
+				{
+					Name:         "test3",
+					Command:      "test3 arg1 arg2",
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     2,
+				},
+			},
+			expectedServicesNameCmdPairs: map[string]string{"test1": "test1 arg1", "test2": "test2"},
+			expectErr:                    false,
+		},
+	}
+	for _, test := range testIO {
+		t.Run(test.name, func(t *testing.T) {
+			desiredVersion := "testversion"
+			cm, err := servicescm.GenerateWithData(servicescm.NamePrefix+desiredVersion,
+				"openshift-windows-machine-config-operator", &test.configMapServices, &[]servicescm.FileInfo{})
+			require.NoError(t, err)
+			data, err := servicescm.Parse(cm.Data)
+			require.NoError(t, err)
+
+			winSvcMgr := winsvc.NewTestMgr(make(map[string]*winsvc.FakeService))
+			c := NewServiceController(context.TODO(), nil, winSvcMgr, "")
+			err = c.Bootstrap(data.Services)
+			if test.expectErr {
+				assert.Error(t, err)
+				return
+			}
+			createdServices, err := getAllFakeServices(winSvcMgr)
+			require.NoError(t, err)
+
+			createdServiceNameCmdPairs := make(map[string]string)
+			for name, createdService := range createdServices {
+				serviceStatus, _ := createdService.Query()
+				assert.Equal(t, svc.Running, serviceStatus.State)
+
+				config, err := createdService.Config()
+				require.NoError(t, err)
+				createdServiceNameCmdPairs[name] = config.BinaryPathName
+			}
```

Comment:
This is repeated code w/ the Reconcile test
it probably should be made into a helper used by both of them

`getFakeServiceCmdNamePairs` or something